### PR TITLE
[irc2] Add port and SSL/TLS support, add IPv6 env variable

### PR DESCRIPTION
**Pull request by codeurimpulsif**
## Goal
- Add port support (defaults to `6667`)
- Add SSL/TLS support (defaults to `0`)
- Add IPv6 environment variable
- Improve documentation
- Remove `host_name` config to avoid localhost IRC monitoring limit

### CODE REVIEW

File: plugins/ircd/irc2 [hunk 1]
```diff
@@ -7,13 +7,18 @@ ircstats  - Plugin to graph data about an IRC network and a single IRC server
...
+- env.PORT port to use, defaults to 6667.
+- env.NICK nickname to use, defaults to munin-$HASH.
+- env.USESSL 0 or 1 values to enable SSL/TLS, defaults to 0.
+- env.USEIPV6 0 or 1 to enable IPv6 use, defaults to 0.
 
 =head1 USAGE
 
 This plugin connects to an IRC server.
 
+It require POE::Component::IRC and POE::Component::SSLify if you use SSL/TLS.
```
Comment (sumpfralle) [inline plugins/ircd/irc2:20]: *It requires* (the `s` was missing)

File: plugins/ircd/irc2 [hunk 2]
```diff
@@ -70,8 +78,10 @@ my $irc = POE::Component::IRC->spawn(
    nick => $nickname,
    ircname => $ircname,
    server => $server,
+   port => $port,
    raw => 0,
-   useipv6 => 0,
+   usessl => $usessl,
```
Comment (sumpfralle) [inline plugins/ircd/irc2:83]: Are you sure, that the string (taken from `$ENV{USESSL}`) is handled properly? Or should you convert it into a numeric value? Is `USESSL=0` really interpreted as *no SSL*?

**Review by sumpfralle — COMMENTED** [review]
Thank you for your contribution!

I added two remarks below.

File: plugins/ircd/irc2 (hunk 1 above)
Comment (codeurimpulsif) [reply plugins/ircd/irc2]: fixed ;)

File: plugins/ircd/irc2 (hunk 2 above)
Comment (codeurimpulsif) [reply plugins/ircd/irc2]: Yes it's really interpreted as *no SSL* as [mentioned in the documentation](https://metacpan.org/pod/POE::Component::IRC#spawn), and maybe I can add and int conversion to be sure?

**Comment by codeurimpulsif** [comment]
Is it ok?

**Comment by sumpfralle** [comment]
Sorry for my late response - I was away for some days ...

Regarding the string `"0"` for `usessl` being interpreted by Perl as `false`: yes, this seems to be one more quirky language feature surprising me (I rarely write Perl). Thank you for clarifying that.

Your contribution looks fine to me. Thank you!

**Pull request merged**